hardware/DenkoviTCPDevices.h

```diff
@@ -86,15 +65,13 @@ class CDenkoviTCPDevices : public CDomoticzHardwareBase, ASyncTCP
 	//boost::shared_ptr<boost::thread> m_thread;
 	std::shared_ptr<std::thread> m_thread;
 	int m_Cmd;
-	bool readingNow = false;
-	bool updateIo = false;
-	bool bFirstTime = true;
-	//_sDenkoviTCPModbusRequest sReq;
-	_sDenkoviTCPModbusRequest *pReq;// = new _sDenkoviTCPModbusRequest;// &sReq;
-	//_sDenkoviTCPModbusResponse sResp;
-	_sDenkoviTCPModbusResponse *pResp;// = new _sDenkoviTCPModbusResponse;// &sResp;
-	uint16_t receivedDataLength = 0;
-	uint16_t transactionCounter = 0;
+	bool m_bReadingNow = false;
+	bool m_bUpdateIo = false;
+	bool m_bFirstTime = true;
+	_sDenkoviTCPModbusRequest *m_pReq;
```

Comment:
Question, why do this have to be a pointer ? Why not

`_sDenkoviTCPModbusRequest m_Req;`

Same for m_Resp

Reply:
I do it in this way because of ASyncTCP returns the respond of request in portions of bytes and for me is easier to handle the respond as a pointer as in this way I just copy received bytes. What do you suggest?

Reply:
Keeping it a member... i am not so happy with new/delete if that is not needed.
Also memcpy and pointers .... i don't like
I think the response of your devices is text based or really hexadecimal byte based?
Still you can use a std::string (.append) maybe ?

Reply:
The response is hexadecimal data and not text based.

Reply:
std::string a = "";
char arr[20] = { 'a',0x00,'b' };
a.append((const char*)arr);//null terminates this std::string
int b = z.length();//length is 1

This does not work for me, but I will look for another way.

Reply:
Is it ok not to be a member of the class and only to be created/deleted (using new/delete) in receive callback?

Reply:
No new/delete please
please have a test with std::string with append,insert etc...
Use it quite a lot when working with small memory